src/daft-logical-plan/src/optimization/rules/split_udfs.rs

```diff
@@ -185,15 +189,21 @@ impl TreeNodeRewriter for TruncateRootActorPoolUDF {
                 }
                 Ok(common_treenode::Transformed::no(node))
             }
+            // TODO: UDFs inside of list.map() can not be split
+            Expr::ScalarFunction(ScalarFunction { udf, .. })
```

Comment:
This is kind of an issue, but I couldn't think of a quick way to fix this by tonights release. Basically, `list_map` is implemented as taking 2 input expressions: the original input and the transformed values. So a `col("data").list.map(lambda x: x + 1)` would be a `list_map(col(data), py_udf(explode(col(data))))`. The `py_udf` is problematic because we want to split it out, but that causes numerous issues that can't be satisfied.

For the time being, I'm just skipping this case, but will discuss with the team on how to better handle this.